spline-shadow/client-ui/src/app/services/lineage/contextual-menu.service.ts

```diff
@@ -29,7 +30,9 @@ export class ContextualMenuService {
     private graphService: GraphService,
     private layoutService: LayoutService,
     private executionPlanControllerService: ExecutionPlanControllerService
-  ) { }
+  ) {
+    executionPlanControllerService.rootUrl = ConfigService.settings.apiUrl
```

Comment:
How can you be sure that `ConfigService.settings` have been set by the time this line is executed? What if there were no settings provided in the `window` variable and they were request by HTTP GET that is still being executed due to a slow internet connection? Shouldn't it be implemented in a reactive manner via a promise or observable?

Reply:
Yeah, but will Angular wait until the promise is resolved?

Reply:
seems so. Ok. thank you